laurent22/joplin
# Desktop: Fixes #10199: Rich text editor: Include "ctrl-click to open" in link tooltips

# Summary

This pull request adds a "ctrl+click to open" prompt to the beginning of tooltips shown in the rich text editor. This prefix is removed before saving. On MacOS, the prompt reads "cmd+click to open". **This pull request has not yet been tested on MacOS**.

**See [this commit](https://github.com/personalizedrefrigerator/joplin/commit/f2b302d9ea31873708bab6a4383a127d3cfd665b#commitcomment-142784614) for an alternate fix.**

Resolves #10199.

# Screenshots

Hovering over a web link shows "ctrl+click to open: http://URL here" or "ctrl+click to open: title here" if the link has a user-set title.
![](https://github.com/laurent22/joplin/assets/46334387/e84ce91f-235f-494c-85bb-af2759364a4b)

Hovering over a note link shows "Ctrl-click to open: joplin://some-note-id-here":
![](https://github.com/laurent22/joplin/assets/46334387/d1b59fdb-e3cb-42f7-874f-162681e91428)


Hovering over a resource link shows "Ctrl-click to open", then the path to the item in Joplin's resource directory:
![](https://github.com/laurent22/joplin/assets/46334387/e2e56b1e-35af-4765-8c9f-007ae06ea3d5)


Titles are also changed in HTML notes:
![](https://github.com/laurent22/joplin/assets/46334387/7531378a-8864-486f-b467-622969957403)


# Possible improvements

- [ ] Show filenames or titles instead of resource and/or note IDs.

# Testing plan

1. Create a new note in the rich text editor with several web links (created with <kbd>ctrl</kbd>+<kbd>k</kbd>).
2. Create a note link with drag-and-drop.
3. Attach a file with the attach button.
4. Verify that all links show a tooltip that starts with "ctrl+click to open" when hovered over.
5. Switch to the markdown editor, then back to the rich text editor.
6. Again, verify that all links show a tooltip that starts with "ctrl+click to open" when hovered over.

This has been tested successfully on Ubuntu 24.04.
